### PR TITLE
Add multi statsd client

### PR DESCRIPTION
**Pull request by Matt-Esch**
I have added a multiplexing statsd client to generate stats with different prefixes.  In doing so I have started to clean up the code to become more consistent with current prevailing styles.

CC: @Raynos

### CODE REVIEW

File: lib/null-statsd-client.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
+'use strict';
+
+var RingBuffer = require('ringbufferjs');
+
+module.exports = NullStatsd;
+
+function NullStatsd(opts, _buffer) {
```
Comment (Raynos) [inline lib/null-statsd-client.js]: This function should be three arguments, `(capacity, prefix, _buffer)`

File: null.js [hunk 2]
```diff
@@ -1,124 +1,9 @@
...
-    this._buffer = new RingBuffer(capacity || 50);
-}
-
-function NullStatsdRecord(type, name, value, delta, time) {
-    this.type = type;
-    this.name = name;
-    this.value = value || null;
-    this.delta = delta || null;
-    this.time = time || null;
+function createNullStatsd(opts) {
```
Comment (Raynos) [inline null.js]: We should put back-compat here.

```js
if (typeof opts === 'number') {
  opts = { capacity: opts }
}
```

File: lib/null-statsd-client.js (hunk 1 above)
Comment (Matt-Esch) [inline lib/null-statsd-client.js]: I made it like this to be interchangeable with the main statsd client

File: null.js (hunk 2 above)
Comment (Matt-Esch) [inline null.js]: Reasonable change to support back compat

File: lib/null-statsd-client.js (hunk 1 above)
Comment (Raynos) [inline lib/null-statsd-client.js]: This seems reasonable. having { prefix: string } here is nice.

**Comment by Matt-Esch** [comment]
@Raynos issues addressed

File: lib/multi-statsd-client.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        clients[i].close();
+    }
+}
+
+function multiCallback(count, cb) {
+    var returnCount = 0;
+    var returned = false;
+
+    return function callbackAfterCount(err) {
+        if (!returned) {
```
Comment (Raynos) [inline lib/multi-statsd-client.js]: nitpick: favor early return.

**Comment by Raynos** [comment]
lgtm. let's get this in.

**Comment by Raynos** [comment]
Suggested version: minor.

**Pull request merged**